src/pages/SearchPage.vue

```diff
@@ -78,40 +87,106 @@ const computedAdvertises = computed(() => {
+    })
+    .sort((a, b) => new Date(b?.id) - new Date(a?.id))
 })
 const computedPromptsAndAdvertises = computed(() => {
-  let i = 0,
-    j = 0
-  let arr = []
+  const map = new Map()
   const promptsLength = computedPrompts.value?.length ?? 0
   const advertisesLength = computedAdvertises.value?.length ?? 0
+
+  let i = 0,
+    j = 0
+
   while (i < promptsLength && j < advertisesLength) {
     if (Math.random() > 0.5) {
-      arr.push(computedPrompts.value[i])
+      if (!map.has(computedPrompts.value[i].id)) {
+        map.set(computedPrompts.value[i].id, computedPrompts.value[i])
+      }
       i++
     } else {
-      arr.push(computedAdvertises.value[j])
+      if (!map.has(computedAdvertises.value[j].id)) {
+        map.set(computedAdvertises.value[j].id, computedAdvertises.value[j])
+      }
       j++
     }
   }
-  if (i < promptsLength) {
-    arr = [...arr, ...computedPrompts.value.slice(i)]
+
+  while (i < promptsLength) {
+    if (!map.has(computedPrompts.value[i].id)) {
+      map.set(computedPrompts.value[i].id, computedPrompts.value[i])
+    }
+    i++
   }
-  if (j < advertisesLength) {
-    arr = [...arr, ...computedAdvertises.value.slice(j)]
+
+  while (j < advertisesLength) {
+    if (!map.has(computedAdvertises.value[j].id)) {
+      map.set(computedAdvertises.value[j].id, computedAdvertises.value[j])
+    }
+    j++
   }
-  return arr
+
+  return Array.from(map.values())
 })
 
 const computedEntries = computed(() => {
   return entryStore.getEntries?.filter((item) =>
     [item.title, item.description, item.author?.displayName].some((str) => str?.toLowerCase().includes(search.value.toLowerCase()))
   )
 })
+const initIntersectionObserver = () => {
+  const options = {
+    root: null,
+    rootMargin: '0px',
+    threshold: 1.0
+  }
+  const observerInstance = new IntersectionObserver(onIntersect, options)
+  observerInstance.observe(observer.value)
+  const scrollTopIntersectionObserver = new IntersectionObserver(onScrollTopIntersect, options)
+  scrollTopIntersectionObserver.observe(scrollTopObserver.value)
+}
+
+function onIntersect(entries) {
+  const [entry] = entries
+  if (entry.isIntersecting) {
+    if (promptStore.hasMore) {
+      promptStore.fetchPrompts(true)
+    }
+  }
+}
+async function onScrollTopIntersect(entries) {
+  const [entry] = entries
+  if (entry.isIntersecting) {
+    loader.value = true
+    promptStore.fetchLatestPrompt().finally(() => (loader.value = false))
+  }
+}
+function computeFetchPromptCount(height, width) {
+  const basePromptCount = 3
+
+  const heightFactor = height / 400
+  const widthFactor = width / 700
+
+  const computedCount = Math.round(basePromptCount + heightFactor * widthFactor)
+
+  return computedCount
+}
+
+onMounted(() => {
+  observer.value.focus()
+  initIntersectionObserver()
+
+  const height = pageRef.value.$el.clientHeight
+  const width = pageRef.value.$el.clientWidth
+  const promptFetchCount = computeFetchPromptCount(height, width)
+
```

Comment:
If you console.log(promptFetchCount) you will se a number that is computed based on your screen size. But when you will open Pinia store you will see that it actually fetches twice the number specified. This probably might be to the loadMore set to true on initial load. And also when you refresh the page with the log of promptFetchCount you will see that each time the number is increased even though the screen size is still the same

<img width="1713" alt="Screenshot 2024-09-09 at 16 50 47" src="https://github.com/user-attachments/assets/cf8bcc2d-acc3-4faf-982b-1caec4466926">
<img width="1721" alt="Screenshot 2024-09-09 at 16 51 37" src="https://github.com/user-attachments/assets/cbaa9930-908a-4c93-82e3-96e8abaa87c1">

Reply:
I will look into it